src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogApiClient.java

```diff
@@ -587,50 +484,38 @@ public boolean validateWebhookIntakeConnection() throws IOException {
     }
 
     @Override
-    public boolean startBuildTrace(BuildData buildData, Run run) {
-        if(this.isWebhookIntakeConnectionBroken()){
-            logger.severe("Unable to start build trace; your client is not initialized properly.");
-            return false;
-        }
-        try {
-            logger.fine("Started build trace");
-            this.webhookBuildLogic.startBuildTrace(buildData, run);
-            return true;
-        } catch (Exception e) {
-            DatadogUtilities.severe(logger, e, "Failed to start build trace");
-            return false;
-        }
+    public TraceWriteStrategy createTraceWriteStrategy() {
+        return new TraceWriteStrategyImpl(new DatadogWebhookBuildLogic(), new DatadogWebhookPipelineLogic(), this::sendSpans);
     }
 
-    @Override
-    public boolean finishBuildTrace(BuildData buildData, Run<?, ?> run) {
-        if(this.isWebhookIntakeConnectionBroken()){
-            logger.severe("Unable to finish build trace; your client is not initialized properly.");
-            return false;
-        }
-        try {
-            logger.fine("Finished build trace");
-            this.webhookBuildLogic.finishBuildTrace(buildData, run);
-            return true;
-        } catch (Exception e) {
-            DatadogUtilities.severe(logger, e, "Failed to finish build trace");
-            return false;
+    private void sendSpans(List<net.sf.json.JSONObject> spans) {
+        for (JSONObject span : spans) {
+            // webhook intake does not support batch requests
+            postWebhook(span.toString());
         }
     }
 
-    @Override
-    public boolean sendPipelineTrace(Run<?, ?> run, FlowNode flowNode) {
-        if(this.isWebhookIntakeConnectionBroken()){
-            logger.severe("Unable to send pipeline trace; your client is not initialized properly");
-            return false;
-        }
-        try {
-            logger.fine("Send pipeline traces.");
-            this.webhookPipelineLogic.execute(run, flowNode);
-            return true;
-        } catch (Exception e) {
-            DatadogUtilities.severe(logger, e, "Failed to send pipeline trace");
-            return false;
+    /**
+     * Posts a given payload to the Datadog Webhook Intake, using the user configured apiKey.
+     *
+     * @param payload - A webhook payload.
+     */
+    private void postWebhook(String payload) {
+        logger.fine("Sending webhook");
+
+        if (this.webhookIntakeConnectionBroken) {
+            throw new RuntimeException("Your client is not initialized properly; webhook intake connection is broken.");
```

Comment:
Nope, the circuit breaker handles it. We log the error and enter a recovery period where we won't attempt sending more spans for a while (this is needed to avoid spamming too many error logs).